feat: add unenroll documentation to mfa docs (#10435)
supabase/supabase: apps/docs/pages/guides/auth/auth-mfa.mdx

}
```

### Add unenrollment flow

An unenrollment flow provides a UI for users to manage and unenroll factors linked to their accounts.
Most applications do so via a factor management page where users can view and unlink selected factors.

> To unenroll a factor, a user simply needs to call `supabase.auth.mfa.unenroll()` with the id of the factor they wish to unenroll.
> For example, we would call: `supabase.auth.mfa.unenroll({factorId: "d30fd651-184e-4748-a928-0a4b9be1d429"})` to unenroll a factor with ID `d30fd651-184e-4748-a928-0a4b9be1d429`.

Reviewer: Part of this got lost in the rebase:
```suggestion
When a user unenrolls a factor, call `supabase.auth.mfa.unenroll()` with the ID of the factor.
For example, call `supabase.auth.mfa.unenroll({factorId: "d30fd651-184e-4748-a928-0a4b9be1d429"})` to unenroll a factor with ID `d30fd651-184e-4748-a928-0a4b9be1d429`.
```

}
```

### Add unenrollment flow

An unenrollment flow provides a UI for users to manage and unenroll factors linked to their accounts.
Most applications do so via a factor management page where users can view and unlink selected factors.

> To unenroll a factor, a user simply needs to call `supabase.auth.mfa.unenroll()` with the id of the factor they wish to unenroll.
> For example, to unenroll the nil UUID we would call: `supabase.auth.mfa.unenroll({factorId: "00000000-0000-0000-0000-000000000000"})`.

Reviewer: I think this could use some clarification between the end-user and the developer/platform:
```suggestion
When a user unenrolls a factor, call `supabase.auth.mfa.unenroll()` with the ID of the factor.
For example, to unenroll the nil UUID, call: `supabase.auth.mfa.unenroll({factorId: "00000000-0000-0000-0000-000000000000"})`.
```

Reviewer 2: Maybe give an example with a real UUID?

Author: Yeah real UUID sounds better - let me change that

#### Example: React

Below is an example that creates a new `UnenrollMFA` component that illustrates
the important pieces of the MFA enrollment flow. Note that a user can only unenroll factors they own and they can only unenroll factors after
first going through the MFA process and attaining the `aal2` level. Here are some points of note:

> - When the component appears on screen, the `supabase.auth.mfa.listFactors()` API is
>   called once to fetch all existing factors together with their details.
> - The existing factors for a user are displayed in a table.
> - Once the user has selected a factor to unenroll they can type in the factorId and hit the `Unenroll` button
>   which creates a confirmation modal.

Reviewer: ```suggestion
- When the component appears on screen, the `supabase.auth.mfa.listFactors()` endpoint
  fetches all existing factors together with their details.
- The existing factors for a user are displayed in a table.
- Once the user has selected a factor to unenroll, they can type in the factorId and click **Unenroll**
  which creates a confirmation modal.
```

<Admonition type="note">
A user can only unenroll a factor after completing the enrollment flow and obtaining
an `aal2` JWT claim.

> </Admonition>

Reviewer: This is pretty similar to the statement in the paragraph above—should get rid of one of them.

```tsx
/**
 * UnenrollMFA shows a simple table with the list of factors together with a button to unenroll.

>  * When a user types in the factorId of the factor that they wish to unenroll

Reviewer: Incomplete sentence?

- `otp` - any one-time password based sign in (email code, SMS code, magic
  link).
- `oauth` - any OAuth based sign in (social login).

> - `mfa/totp` - a TOTP additional factor.

Reviewer 2: I think this is `mfa/totp` no?

Author: oh I think it's just `totp` 

https://github.com/supabase/gotrue/blob/master/models/factor.go#L39
